src/v/security/gssapi_authenticator.cc

```diff
@@ -208,6 +213,13 @@ ss::future<result<bytes>> gssapi_authenticator::authenticate(bytes auth_bytes) {
             _impl->reset();
             return principal;
         });
+        _session_expiry.reset();
+        auto lifetime = _impl->lifetime();
```

Comment:
yeah that's a great point. I'd not picked up on the pattern. offhand, it looks like maybe the principal one is wrapped up that way due to the `reset` call, which looks like it could block. Not really sure.

Reply:
went ahead and made this change. I'm still curious to know whether it's necessary, but seems like it can't hurt.

Reply:
Because reading and writing from different threads is a data race. The submit on the worker synchronizes that.